File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/CurrentUserCodySubscription.kt
package com.sourcegraph.cody.agent.protocol

import com.google.gson.annotations.SerializedName
import java.util.*

data class CurrentUserCodySubscription(
    val status: Status,
    val plan: Plan,
    val applyProRateLimits: Boolean,
    val currentPeriodStartAt: Date,
    val currentPeriodEndAt: Date,
)

enum class Plan {
  @SerializedName("PRO") PRO,
  @SerializedName("FREE") FREE
}

enum class Status {
  @SerializedName("ACTIVE") ACTIVE,
  @SerializedName("PAST_DUE") PAST_DUE,
  @SerializedName("UNPAID") UNPAID,
  @SerializedName("CANCELED") CANCELED,
  @SerializedName("TRIALING") TRIALING,
  @SerializedName("PENDING") PENDING,
  @SerializedName("OTHER") OTHER,
}

File: agent/bindings/csharp/CurrentUserCodySubscription.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class CurrentUserCodySubscription
  {
    [JsonProperty(PropertyName = "status")]
    public string Status { get; set; }
    [JsonProperty(PropertyName = "plan")]
    public string Plan { get; set; }
    [JsonProperty(PropertyName = "applyProRateLimits")]
    public bool ApplyProRateLimits { get; set; }
    [JsonProperty(PropertyName = "currentPeriodStartAt")]
    public Date CurrentPeriodStartAt { get; set; }
    [JsonProperty(PropertyName = "currentPeriodEndAt")]
    public Date CurrentPeriodEndAt { get; set; }
  }
}
